src/Microsoft.AspNet.Authentication.OpenIdConnect/OpenIdConnectMiddleware.cs

```diff
@@ -91,10 +91,7 @@ public class OpenIdConnectMiddleware : AuthenticationMiddleware<OpenIdConnectOpt
                 throw new ArgumentException("Options.SignInScheme is required.");
             }
 
-            if (Options.HtmlEncoder == null)
-            {
-                Options.HtmlEncoder = services.GetHtmlEncoder();
-            }
+            HtmlEncoder = services.GetHtmlEncoder();
```

Comment:
Fair enough I'll make that change.  Slightly unrelated question, do you think we need to all of these duplicate null arguement checks?  My preference is to nuke them from the child ctors if the base ctor is already doing the checks.  We have so many overloads that each are doing the same null check since the NotNull tool put them EVERYWHERE.